src/marshmallow/fields.py

```diff
@@ -467,24 +519,33 @@ def schema(self):
         if not self._schema:
             # Inherit context from parent.
             context = getattr(self.parent, "context", {})
-            if isinstance(self.nested, SchemaABC):
-                self._schema = self.nested
+            if callable(self.nested) and not isinstance(self.nested, type):
```

Comment:
Alternatively

```py
if inspect.isfunction(self.nested):
```

Reply:
That would break for callable objects (`__call__`). May not be a typical use case, but I think we might as well support it.

Reply:
Huh, just discovered that `isfunction` also returns False for `partial`s 🤔 

```python

In [1]: from functools import partial

In [2]: import inspect

In [3]: def foo(x):
   ...:     return 42
   ...:

In [4]: foo2 = partial(foo, x=24)

In [5]: inspect.isfunction(foo2)
Out[5]: False
```

Reply:
Surprising, indeed.